pkg/handlers/pachyderm_handler.go

```diff
@@ -0,0 +1,274 @@
+		if err != nil {
+			return kvcv1.Volume{
+				ID:      vc.ID,
+				Message: fmt.Sprintf("error while parsing timeout for data download: %v", err),
+			}
+		}
+	}
+
+	nodeClient := getK8SResourceClientFromPlural(h.k8sResourceClients, "nodes")
+	nodeList, err := nodeClient.List(ns, map[string]string{})
+	if err != nil {
+		return kvcv1.Volume{
+			ID:      vc.ID,
+			Message: fmt.Sprintf("error getting node list: %v", err),
+		}
+	}
+
+	// If number of nodes < replicas, then return immediately.
+	if len(nodeList) < vc.Replicas {
+		return kvcv1.Volume{
+			ID: vc.ID,
+			Message: fmt.Sprintf("replicas [%v] greater than number of nodes [%v]",
+				vc.Replicas, len(nodeList)),
+		}
+	}
+
+	vc.Options["recursive"] = ""
+	if strings.HasSuffix(vc.Options["inputPath"], "/") {
+		vc.Options["recursive"] = "-r"
+	}
+
+	kvcNames := []string{}
+	podClient := getK8SResourceClientFromPlural(h.k8sResourceClients, "pods")
+	kvcDataPathSuffix := fmt.Sprintf("%s%s", kvcNamePrefix, uuid.NewUUID())
+	for i := 0; i < vc.Replicas; i++ {
+		kvcName := fmt.Sprintf("%s%s", kvcNamePrefix, uuid.NewUUID())
+		kvcNames = append(kvcNames, kvcName)
+
+		err = podClient.Create(ns, struct {
+			kvcv1.VolumeConfig
+			metav1.OwnerReference
+			NS                  string
+			KVCName             string
+			KVCStorageClassName string
+			PVType              string
+			KVCOptions          map[string]string
+		}{
+			vc,
+			controllerRef,
+			ns,
+			kvcName,
+			"kvc",
+			"",
+			map[string]string{
+				"path": fmt.Sprintf("%s/%s", vc.Options["dataPath"], kvcDataPathSuffix),
+			},
+		})
+
+		if err != nil {
+			return kvcv1.Volume{
+				ID:      vc.ID,
+				Message: fmt.Sprintf("error during sub-resource [%s] creation: %v", podClient.Plural(), err),
+			}
+		}
+	}
+
+	usedNodeNames := []string{}
+	for _, kvcName := range kvcNames {
+		err := waitForPodSuccess(podClient, kvcName, ns, timeout)
+		if err != nil {
+			return kvcv1.Volume{
+				ID: vc.ID,
+				// TODO(balajismaniam): append pod logs to this message if possible.
+				Message: fmt.Sprintf("error during data download using pod [name: %v]: %v", kvcName, err),
+			}
+		}
+
+		podObj, err := podClient.Get(ns, kvcName)
+		if err != nil {
+			return kvcv1.Volume{
+				ID:      vc.ID,
+				Message: fmt.Sprintf("error getting pod [name: %v]: %v", kvcName, err),
+			}
+		}
+
+		pod, ok := podObj.(*corev1.Pod)
+		if !ok {
+			return kvcv1.Volume{
+				ID:      vc.ID,
+				Message: fmt.Sprintf("object returned from podclient.Get() is not a pod"),
+			}
+		}
+
+		usedNodeNames = append(usedNodeNames, pod.Spec.NodeName)
+
+		node, err := nodeClient.Get("", pod.Spec.NodeName)
+		if err != nil {
+			return kvcv1.Volume{
+				ID:      vc.ID,
+				Message: fmt.Sprintf("could not get node %s, error: %v", pod.Spec.NodeName, err),
+			}
+		}
+		// update nodes with the correct label
+		err = updateNodeWithLabels(nodeClient, node.(*corev1.Node), []string{fmt.Sprintf("%s/%s-%s", kvcv1.GroupName, ns, controllerRef.Name)}, "add")
```

Comment:
Can you change the node label key to https://github.com/kubeflow/experimental-kvc/blob/master/pkg/handlers/s3_handler.go#L156?